packages/cli/src/commands/versions/bump.ts

```diff
@@ -199,8 +217,12 @@ export default async (cmd: Command) => {
 
     console.log();
 
-    await bumpBackstageJsonVersion();
-
+    // Do not update backstage.json when upgrade patterns are used.
+    if (pattern === DEFAULT_PATTERN_GLOB) {
```

Comment:
This is gonna cause issues for people doing things like `--pattern '@{roadiehq,backstage}/*'`. Perhaps we could log something along the lines of "because custom pattern we no bump backstage.json, but latest version is x.y.z"

Reply:
Added log line below